src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -0,0 +1,217 @@
+        this.tasksOps = tasksOps;
+        this.workerOps = workerOps;
+        this.fallbackWorker = fallbackWorker;
+        this.fallbackPayment = fallbackPayment;
+
+        //create hashmap of platforms
+        platforms = crowdPlatforms.stream()
+                .collect(Collectors.toMap(Platform::getName, Function.identity()));
+        //clear database
+        platformOps.deleteAllPlatforms();
+        //update database
+        platforms.forEach((s, platform) -> {
+            PlatformRecord rec = new PlatformRecord();
+            rec.setName(platform.getName());
+            rec.setNeedsEmail(false);
+
+            /* platform does not handle payment, email is needed for internal payment */
+            if (!platform.getPayment().isPresent())
+                rec.setNeedsEmail(true);
+            /* if platform cannot identify worker, we need to do that with a email adress */
+            if (!platform.getWorker().isPresent())
+                rec.setNeedsEmail(true);
+
+            rec.setRenderCalibrations(platform.isCalibsAllowd());
+
+            platformOps.createPlatform(rec);
+        });
+    }
+
+    /**
+     * Will get you the instance of a platform interface of a platform, this instance is the same for all calls
+     * @param name The name of the instance to use
+     * @return The optional crowd platform instance
+     */
+    public Optional<Platform> getPlatform(String name) {
+        return Optional.ofNullable(platforms.get(name));
+    }
+
+    /**
+     * Will return the Worker interface which should be used to identify workers for the given platform
+     *
+     * @param name The name of the platform
+     * @return The interface used to identify a worker
+     */
+    public Optional<Worker> getWorker(String name) {
+        return getPlatform(name).map(platform -> platform.getWorker().orElse(fallbackWorker));
+    }
+
+    /**
+     * Will return the payment service which should be used for the given platform
+     * If there is no Platform with the given name None is returned.
+     *
+     * @param name The name of the platform to use
+     * @return The interface used for payment
+     */
+    public Optional<Payment> getPlatformPayment(String name) {
+        return getPlatform(name).map(platform -> platform.getPayment().orElse(fallbackPayment));
+    }
+
+    /**
+     * Publish the given experiment on the platform.
+     * The method will update the database with the new public task
+     *
+     * @param name The name of the platform
+     * @param experiment The experiment to publish
+     * @return None if the platform does not exist
+     */
+    public Optional<CompletableFuture<Boolean>> publishTask(String name, Experiment experiment) throws TaskOperationException {
+        if (tasksOps.searchTask(name,experiment.getId()).isPresent())
+            throw new TaskOperationException("Experiment is already published!");
```

Comment:
if you publish a task twice then on the if clause would fail for both requests and therefore not throw an exception. I don't what happens then with the whole system. Therefore you don`t check mutable things like that if you change them afterwards